proteuslib/tools/initialization_analyzer.py

```diff
@@ -0,0 +1,112 @@
+###############################################################################
+# ProteusLib Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/nawi-hub/proteuslib/"
+#
+###############################################################################
+
+import weakref
+import pyomo.environ as pyo
+
+from idaes.core.util import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from .parameter_sweep import parameter_sweep as _parameter_sweep
+
+
+class _FlowsheetInitializationAnalyzer:
+
+    def __init__(self, flowsheet, initializer, solver=None):
+        self.model = flowsheet
+        self._initializer = initializer
+        self._store_initial_values()
+
+        if hasattr(self.model, '_fsa'):
+            raise RuntimeError(f"Constructed model has attribute _fsa")
+        self.model._fsa = weakref.ref(self)
```

Comment:
Just so I understand the usage here-- by using `weakref`, I am guessing this helps clears some memory when the function returns output, but I'm not 100% clear on the logic here (lines 31 - 32). Will anything tied to `self.simulate_parameter_sweep` be stripped from `self.model` at the end of the function call?